types/index.d.ts

```diff
@@ -0,0 +1,87 @@
+// Minimum TypeScript Version: 3.2
+import {Plugin} from 'unified'
+
+/**
+ * Wrap a document around a fragment.
+ */
+declare const document: Plugin<[document.Options?]>
+
+declare namespace document {
+  interface Options {
+    /**
+     * Text to use as title
+     *
+     * default: name of file, if any
+     */
+    title?: string
+
+    /**
+     * Natural language of document.
+     *
+     * should be a [BCP 47](https://tools.ietf.org/html/bcp47) language tag.
+     *
+     * @default 'en'
+     */
+    language?: string
+
+    /**
+     * [Doctype](https://github.com/wooorm/doctype) to use.
+     *
+     * @default '5'
+     */
+    doctype?: string
+
+    /**
+     * CSS to include in `head` in `<style>` elements.
+     *
+     * @default []
+     */
+    style?: string | string[]
+
+    /**
+     * Links to stylesheets to include in `head`
+     *
+     * @default []
+     */
+    css?: string | string[]
+
+    /**
+     * Metadata to include in `head`.
+     *
+     * Each object is passed as
+     * [`properties`](https://github.com/syntax-tree/hastscript#hselector-properties-children)
+     * to [`hastscript`](https://github.com/syntax-tree/hastscript) with a
+     * `meta` element.
+     *
+     * @default []
+     */
+    meta?: Record<string, any> | Array<Record<string, any>>
```

Comment:
Based off the tests I think meta takes string keys and values
https://github.com/rehypejs/rehype-document/blob/498a1669121f9cbd3241f550712ab5654956fec6/test.js#L162-L165
```suggestion
    meta?: Record<string, string> | Array<Record<string, string>>
```

Reply:
I based these types on the existing type of `Properties` in `@types/hast`.

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/hast/index.d.ts#L58

This seems correct reading the definition of properties in hast https://github.com/syntax-tree/hast#properties

Reply:
@wooorm may be able to weigh in with more detail,
looking over the spec I tend to interpret it to mean primitives and arrays of primitives (`string | string[] | number | boolean`), over `any` which could be `Promise`, `Buffer`, or any number of complex types.
If that interpretation is correct, I'll open a PR to `hast` to update that.

Reply:
`string`, `number`, `true`, `Array.<string|number>` are all values that are supported. `boolean[]` doesn’t make sense. Arrays of mixed strings and numbers does make a bit of sense, I think we reflect that for `<x coords="123, asd, 456"></x>`

`false`, `null`, `undefined` are also fine.

`NaN` is probably an error by a user, but we treat it as `false`, `null`, `undefined` in utilities.

Reply:
Arrays of numbers are uncommon though. Here’s where we pull from: https://github.com/wooorm/property-information/blob/ac3452348fc694ec8bb2d3f0080b25a9dfa7f8e2/lib/html.js#L53

Reply:
I don’t think this package is the proper place to define these types. Should I just change the type to `Properties` from `@types/hast`? Strictening types would be an upstream issue then.

Reply:
I think so, yeah

Reply:
upstream change https://github.com/DefinitelyTyped/DefinitelyTyped/pull/44341